### PR TITLE
feature/simplify-build

### PR DESCRIPTION
**Pull request by JacksonMaxfield**
Switch to CIBuildWheel for wheel building and clean up the github actions

Most of this work was done by @toloudis so I verbal "approve" of this PR.

### CODE REVIEW

File: .github/workflows/py-build-main.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  CIBW_SKIP: cp36-* cp310-* *-win32 pp*
+
+name: Python Build Main
+jobs:
+  python-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-latest, windows-2019, macOS-10.15, macOS-11.0]
```
Comment (toloudis) [inline .github/workflows/py-build-main.yml:20]: I wonder if this one could be windows-latest here

File: .github/workflows/py-build-pr.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+name: Python Build PR
 jobs:
   python-test:
     runs-on: ${{ matrix.os }}
     strategy:
       max-parallel: 6
       matrix:
         python-version: [3.7, 3.8, 3.9]
-        os: [ubuntu-latest, windows-2019, macos-10.15]
+        os: [ubuntu-latest, windows-2019, macos-10.15, macos-11.0]
```
Comment (toloudis) [inline .github/workflows/py-build-pr.yml:10]: same as in py-build-main, is windows-latest preferable?

**Review by toloudis — APPROVED** [review]

File: .github/workflows/py-build-main.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    steps:
+      - uses: actions/checkout@v1
```
Comment (heeler) [inline .github/workflows/py-build-main.yml:22]: why actions/checkout@v1 here and actions/checkout@v2 below?

File: .github/workflows/py-build-main.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      - name: Lint with flake8
+        run: |
+          flake8 aicspylibczi --count --verbose --max-line-length=127 --show-source --statistics
+
+  build_sdist:
+    name: Build SDist
+    needs: [python-test, lint]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (heeler) [inline .github/workflows/py-build-main.yml:63]: other

**Review by heeler — APPROVED** [review]
So there are a few things to make consistent ie the action to checkout being @v1 or @v2 but not both, if @v2 now supports submodules then run with that. 
The updates to the script all look sane and reasonable. 
I've been swamped with end of quarter goals but I would like to suggest some further improvements that I've been messing with in my day job that should make the build cleaner:
1) CMake: [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) allows you to make the submodules build dependencies rather than submodules which seems much cleaner to me. (It does require a relatively modern cmake be present)
2) If we switch to pyproject.toml rather than setup.py it basically just works to trigger the build, not nearly as complex as the setup.py method, note if you want to do local installs you should leave a setup.py "shim" to enable that, it's well documented.  The cost of going to toml is that it requires the latest and greatest pip but that seems like not a big deal unless I'm missing something.

File: .github/workflows/py-build-main.yml (hunk 3 above)
Comment (JacksonMaxfield) [reply .github/workflows/py-build-main.yml:22]: because :shrug:

**Pull request merged**